src/main/extension/extension_install.cpp

```diff
@@ -347,46 +349,82 @@ static unique_ptr<ExtensionInstallInfo> InstallFromHttpUrl(DBConfig &config, con
 	auto hostname_without_http = no_http.substr(0, next);
 	auto url_local_part = no_http.substr(next);
 
-	auto url_base = "http://" + hostname_without_http;
-	duckdb_httplib::Client cli(url_base.c_str());
-	if (http_logger) {
-		cli.set_logger(http_logger->GetLogger<duckdb_httplib::Request, duckdb_httplib::Response>());
-	}
-
-	duckdb_httplib::Headers headers = {
-	    {"User-Agent", StringUtil::Format("%s %s", config.UserAgent(), DuckDB::SourceID())}};
-
 	unique_ptr<ExtensionInstallInfo> install_info;
 	{
 		auto fs = FileSystem::CreateLocal();
 		if (fs->FileExists(local_extension_path + ".info")) {
 			install_info = ExtensionInstallInfo::TryReadInfoFile(*fs, local_extension_path + ".info", extension_name);
 		}
+	}
+
+	auto url_base = "http://" + hostname_without_http;
+	static constexpr const idx_t MAX_RETRY_COUNT = 3;
+	static constexpr uint64_t RETRY_WAIT_MS = 100;
+	static constexpr double RETRY_BACKOFF = 4;
+	idx_t retry_count = 0;
+	duckdb_httplib::Result res;
+	while (true) {
```

Comment:
Perhaps it makes sense to add a TODO in the code related to the need to unify this with the httpfs extension code?

Reply:
Sure, added